src/node_file.cc

```diff
@@ -138,9 +124,27 @@ void FSReqWrap::Resolve(Local<Value> value) {
   MakeCallback(env()->oncomplete_string(), arraysize(argv), argv);
 }
 
+void FSReqWrap::Init(const char* syscall,
+                     const char* data,
+                     enum encoding encoding,
+                     Ownership ownership) {
+  CHECK_EQ(data_, nullptr);  // Initialize only once
+  syscall_ = syscall;
+  encoding_ = encoding;
+
+  const bool copy = (data != nullptr && ownership == Ownership::COPY);
+  if (copy) {
+    // Setting buffer_ lets us know that we'll need to free on cleanup
+    data_ = buffer_ = strdup(data);
```

Comment:
`strdup()` is almost certainly wrong.  That only works for zero-terminated strings whereas `data` can be arbitrary binary data, as far as I can tell.

Reply:
Ok yeah, looking at it, I *think* that current uses of this just end up working with strdup by accident. You're right tho, the code appears to have been originally written to support arbitrary binary data. Will change that back